c10/core/TensorImpl.cpp

```diff
@@ -113,6 +113,47 @@ bool TensorImpl::has_storage() const {
   return storage_;
 }
 
+void TensorImpl::set_memory_format_tag(MemoryFormat memory_format) {
+  AT_ASSERT(
+      !is_variable()); // TODO: remove this when Variable and Tensor are merged
+  memory_format_tag_ = memory_format;
+}
+
+bool TensorImpl::maybe_as_channels_last() {
+  AT_ASSERT(
+      !is_variable()); // TODO: remove this when Variable and Tensor are merged
+  if (dim() == 4 && is_contiguous_) {
```

Comment:
I guess my `torch.select` concern is slightly moot, given the current implementation, because there is no support for tagging 3-D tensors as channels-last